src/client/producer.rs

```diff
@@ -300,6 +300,16 @@ impl BatchProducerBuilder {
     }
 }
 
+/// The result of the produce call.
+#[derive(Debug, Default)]
+pub struct ProduceResult {
```

Comment:
```suggestion
#[non_exhaustive]
pub struct ProduceResult {
```

makes it easier to extend this later w/o introducing breaking changes.